src/AdoPat/IPatCache.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AdoPat
+{
+    using Microsoft.VisualStudio.Services.DelegatedAuthorization;
+
+    /// <summary>
+    /// A simple cache mapping <see cref="string"/> to <see cref="PatToken"/> backed by secure storage.
+    /// </summary>
+    public interface IPatCache
+    {
+        /// <summary>
+        /// Get a <see cref="PatToken"/> from the cache.
+        /// </summary>
+        /// <param name="key">The key for the target entry.</param>
+        /// <returns>The target value.</returns>
+        PatToken GetPat(string key);
+
+        /// <summary>
+        /// Put a <see cref="PatToken"/> into the cache. May overwrite existing values.
+        /// </summary>
+        /// <param name="key">The key for this entry.</param>
+        /// <param name="patToken">The value for this entry.</param>
+        void PutPat(string key, PatToken patToken);

```

Comment:
Nit - The name sounds a bit off. PutPat, GetPat. I also notice you're using the term ```PatToken``` and Pat to refer to the same PAT.  Since T in PAT stands for token, should be rename PatToken to be PersonalAccessToken? 

As for these methods, my personal choice would be to either name them as ```GetPAT```, ```PutPAT``` or  simply ```GetToken```, ```PutToken``` since we are already in the context of PATCache here

Reply:
I only used `patToken` because the class is called `PatToken` and I don't have control over that since it's a type that comes from the Visual Studio ADO API library. Personally, I think it's a silly name as well. Seems kinda like ATM Machine or PIN Number.

As for the method names, I was under the impression that in C# Pascal case names like `Pat` were preferred to uppercase names like `PAT` even for initialisms or acronyms. I don't like `GetToken` or `PutToken` because there's enough ambiguity in this code base with multiple types of tokens.

However, as you say, we're already in the context of a `PatCache` here, so I'd happily go for `PatCache.Get()` or `PatCache.Put()`, which I think would be simpler.

Reply:
Well that is silly.

Reply:
`PatCache.GetToken()` should work and be consistent. Even if we use `GetToken` in multiple places for different types of token fetching, the class names are distinct.